promql/quantile.go

```diff
@@ -106,3 +106,33 @@ func quantile(q model.SampleValue, buckets buckets) float64 {
 	}
 	return bucketStart + (bucketEnd-bucketStart)*float64(rank/count)
 }
+
+// qauntile calculates the given quantile of a slice of floats.
+//
+// The slice will be sorted.
+// If 'buckets' has zero elements, NaN is returned.
+// If q<0, -Inf is returned.
+// If q>1, +Inf is returned.
+func quantile(q float64, values []float64) float64 {
+	if len(values) == 0 {
+		return math.NaN()
+	}
+	if q < 0 {
+		return math.Inf(-1)
+	}
+	if q > 1 {
+		return math.Inf(+1)
+	}
+	sort.Float64s(values)
```

Comment:
Yes, that's the case against having model.SampleValue at all. Not a fan, as said.

But in this PR, we get things like 
```
value   model.SampleValue
values  []float64
```

or compare the signaruter of `bucketQuantiles` with `quantiles`. As said, my brain starts to waste cycles thinking what the fundamental difference is while reading the code. It distracts from what the code is actually doing.